src/arrayMethodSort.js

```diff
@@ -4,8 +4,25 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compares =
+      compareFunction !== undefined
+        ? compareFunction
+        : (a, b) => String(a) > String(b);
+
+    // Пузырьковые метод сортировки
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - i - 1; j++) {
+        const temp = this[j];
+
+        if (compares(temp, this[j + 1]) > 0) {
```

Comment:
The comparison function `compares` should return a number. Therefore, the condition should be `if (compares(temp, this[j + 1]) > 0)` to properly use the comparison function's result.